spec/classes/init_spec.rb

```diff
@@ -613,4 +613,68 @@
       })
     }
   end
+
+  describe 'parameter lookup' do
+    context 'set to default value' do
+      context 'on OpenBSD' do
+        let :facts do
+          { :operatingsystem => 'OpenBSD' }
+        end
+        it {
+          should contain_file('dnsclient_resolver_config_file').with_content(
+%{# This file is being maintained by Puppet.
+# DO NOT EDIT
+options rotate timeout:1
+nameserver 8.8.8.8
+nameserver 8.8.4.4
+})
+        }
+      end
+      context 'on non-OpenBSD' do
+        it {
+          should contain_file('dnsclient_resolver_config_file').with_content(
+%{# This file is being maintained by Puppet.
+# DO NOT EDIT
+options rotate timeout:1
+nameserver 8.8.8.8
+nameserver 8.8.4.4
+})
+        }
+      end
+    end
+
+    context 'set to a valid array' do
+      let :params do
+        { :lookup => ['file', 'bind'] }
+      end
+      context 'on OpenBSD' do
+        let :facts do
+          { :operatingsystem => 'OpenBSD' }
+        end
+        it {
+          should contain_file('dnsclient_resolver_config_file').with_content(
+%{# This file is being maintained by Puppet.
+# DO NOT EDIT
+options rotate timeout:1
+nameserver 8.8.8.8
+nameserver 8.8.4.4
+lookup file bind
+})
+        }
+      end
+      context 'on non-OpenBSD' do
+        it { should_not compile }
```

Comment:
This would mean that you have a `let :facts` set with `:operatingsystem => 'not-openbsd'`